drkafka/src/main/java/com/pinterest/doctorkafka/KafkaClusterManager.java

```diff
@@ -657,10 +677,35 @@ public void handleUnderReplicatedPartitions(List<PartitionInfo> initialUrps,
    * a broker is dead or not. This method only returns true when we are sure that the broker
    * is not available.
    */
-  private boolean isDeadBroker(String host, int brokerId) {
+  private boolean isDeadBroker(String host, int brokerId, TopicPartition tp) {
     if (OperatorUtil.pingKafkaBroker(host, 9092, 5000)) {
       LOG.debug("Broker {} is alive as {}:9092 is reachable", brokerId, host);
-      return false;
+      if (tp == null) {
+        KafkaBroker broker = kafkaCluster.brokers.get(brokerId);
+        if (broker != null) {
+          // Try to fetch some data from that broker
+          List<TopicPartition> tps = broker.getLeaderTopicPartitions();
+          if (tps == null || tps.isEmpty()) {
+            tps = broker.getFollowerTopicPartitions();
+          }
+          if (tps != null && !tps.isEmpty()) {
+            tp = tps.get(0);
+          }
+        }
+      }
+      if (tp == null) {
+        LOG.warn("No topic/partitions found for broker {}", brokerId);
+        return false;
+      } else {
+        if (OperatorUtil.fetchData(host, 9092, tp.topic(), tp.partition())) {
```

Comment:
shall we merge  line 699 and line 700 to reduce one level of indention for line 700 -707?

Reply:
Done.